ai_genomics/analysis/integrated_emergence/make_integrated_emergence.py

```diff
@@ -0,0 +1,359 @@
+# Generate integrated analysis of evolution of genomics activity
+
+from ai_genomics.getters.openalex import (
+    get_openalex_ai_genomics_works,
+    get_openalex_entities,
+    get_openalex_ai_genomics_works_entity_groups,
+)
+from ai_genomics.getters.patents import (
+    get_ai_genomics_patents,
+    get_patent_ai_genomics_entity_groups,
+)
+from ai_genomics.getters.gtr import (
+    get_ai_genomics_project_table,
+    get_gtr_ai_genomics_project_entity_groups,
+)
+from ai_genomics.getters.entities import get_entity_cluster_lookup
+
+import altair as alt
+import logging
+from toolz import pipe
+from itertools import chain
+from collections import Counter
+from typing import List, Dict
+import pandas as pd
+import numpy as np
+from ai_genomics.utils.save_plotting import AltairSaver
+from ai_genomics.utils.plotting import configure_plots
+
+
+# Functions
+def make_entity_cluster_names(level: int = 100, top_names: int = 3) -> Dict:
+    """Makes the names for entity clusters using openalex entity frequencies
+    Args:
+        level: the clustering level to use
+        top_names: top number of entities to use in the cluster name
+
+    Returns:
+        A dict between cluster ids and names
+    """
+
+    return (
+        pipe(
+            get_openalex_entities(),
+            lambda _dict: {k: v for k, v in _dict.items() if (len(v) > 0)},
+            lambda _dict: {k: [el[0] for el in v] for k, v in _dict.items()},
+            lambda _dict: pd.Series(Counter(chain(*[v for v in _dict.values()]))),
+        )
+        .to_frame()
+        .rename(columns={0: "frequency"})
+        .assign(cluster=lambda df: df.index.map(get_entity_cluster_lookup(level)))
+        .reset_index(drop=False)
+        .groupby("cluster")
+        .apply(
+            lambda df: "_".join(
+                df.sort_values("frequency", ascending=False)["index"][:top_names]
+            ).lower()
+        )
+        .to_dict()
+    )
+
+
+def make_doc_meta_dict(data_source: str) -> Dict:
+    """Makes a lookup between doc ids and relevant metadata
+    (years in the case of openalex and gtr, family id and pub year in the case of patents)
+
+    Args:
+        data_source: the data source to use
+    """
+
+    if data_source == "openalex":
+        return (
+            get_openalex_ai_genomics_works()
+            .query("ai_genomics==True")
+            .set_index("work_id")["publication_year"]
+            .to_dict()
+        )
+
+    if data_source == "patents":
+        return (
+            get_ai_genomics_patents()
+            .set_index("publication_number")
+            .assign(
```

Comment:
you could convert filing_date to a datetime then just take df['filing_date].dt.year but this also works